### PR TITLE
Containerize FHIR Converter

### PR DESCRIPTION
**Pull request by nickclyde**
Resolves [this ticket](https://reportstream-service-request.clickup.com/t/2tf6v3h)

Still need to do a bit of cleanup, but this is working inside a container!

image

### CODE REVIEW

File: cloud-run/fhir-converter/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+WORKDIR /app
+
+COPY requirements.txt .
+RUN pip install -r requirements.txt
+
+COPY main.py .
+
+EXPOSE 8000
+CMD uvicorn main:api --host 0.0.0.0 --port 8000
```
Comment (DanPaseltiner) [inline cloud-run/fhir-converter/Dockerfile:25]: I believe the GCP Clound run default is to use port 8080. Could we switch to 8080?

**Review by DanPaseltiner — COMMENTED** [review]
Looks good to me so far. I think we are missing unit testing currently. Super exciting to see this coming together!

File: cloud-run/fhir-converter/main.py [hunk 2]
```diff
@@ -1,16 +1,37 @@
...
+
+
+class FhirConverterInput(BaseModel):
+    """
+    Input parameters for the FHIR Converter.
+    """
+
+    input_data: str
+    input_type: str
+    root_template: str
```
Comment (DanPaseltiner) [inline cloud-run/fhir-converter/main.py:18]: I really like this use of Pydantic, it actually inspired me to get something similar going for the `upload_fhir_bundle` function. Could we add some additional specificity/validation here? For example, while 'input_data' will be a string it should also be valid JSON. Also, can probably treat `input_type` as an enum. We might be able to do the same for `root_template` but we will need to think about templates being added and removed in the future.

Comment (nickclyde) [reply cloud-run/fhir-converter/main.py:18]: I like the idea of using enums, I'll add that in! But is it true that `input_data` should be valid JSON? The [sample input HL7 data I used](https://github.com/microsoft/FHIR-Converter/blob/main/data/SampleData/Hl7v2/ADT-A01-01.hl7) looks like pipe delimited data?

Comment (DanPaseltiner) [reply cloud-run/fhir-converter/main.py:18]: Yes, you are totally right! I was still lost thinking in terms of the upload function which takes FHIR as input. Our input data here with either being pipe delimitated Hl7 or XML (CCDA for the ECRs).

**Comment by nickclyde** [comment]
Enum validations work nicely!
Screen Shot 2022-08-16 at 11 38 37 AM

File: cloud-run/fhir-converter/FHIR-Converter-Installation-And-Usage-Guide.md [hunk 3]
```diff
@@ -1,6 +1,22 @@
...
+  1. Start a container by running:  
+  `docker run -p 8080:8080 fhir-converter`
+  1. Using curl or a REST client like Postman or Insomnia, make a POST request to http://localhost:8080/convert-to-fhir with a request body:
+  ```
+  {
+    "input_data": "VALID_INPUT_DATA",
+    "input_type": "hl7v2",
+    "root_template": "ADT_A01",
+  }
+  ```
```
Comment (DanPaseltiner) [inline cloud-run/fhir-converter/FHIR-Converter-Installation-And-Usage-Guide.md:18]: Major nit, but could make it clear that this is an example request. Calling `VALID_INPUT_DATA` does this in part, but I'm concerned a reader might assume that `"input_data"` must be "hl7v2" and that `"root_template"` must be "ADT_A01".

**Review by DanPaseltiner — APPROVED** [review]
LGTM! Thanks for your work on this.

File: cloud-run/fhir-converter/FHIR-Converter-Installation-And-Usage-Guide.md (hunk 3 above)
Comment (nickclyde) [reply cloud-run/fhir-converter/FHIR-Converter-Installation-And-Usage-Guide.md:18]: Good call, will make this more clear

**Comment by celestemespinoza** [comment]
Task linked: [CU-2tf6v3h Create a containerized application around the MSFT FHIR Converter](https://app.clickup.com/t/2tf6v3h)

**Pull request merged**